### DISCUSSION TITLE
Easy mod increases Star Rating in osu!catch

### DISCUSSION THREAD

**Discussion by Tenexxt** [ppy/osu#29544] [Q&A]
### Type

Game behaviour

### Bug description

I noticed while playing !CtB that the "Easy" mod increases SR instead of lowering it.
After some more looking into it i found out that this happens on higher difficulties (Platter and higher). I think the reason for this happening is when using the Easy mod, the game erases hyperdashes in the map, resulting in making some jumps very hard, which also results in making the SR higher. This doesn't happen always, but in some cases this can greatly increase the SR, even tho it should result in making it lower.

Easy mod on: 
![image](https://github.com/user-attachments/assets/c446c28c-c8a9-4176-a31c-37be35730814)

No mod:
![image](https://github.com/user-attachments/assets/4cb95d1c-76f8-4cbb-9bad-4a395750290f)

The map from the example can be found [here](https://osu.ppy.sh/beatmapsets/1772651#fruits/3629285) (its the platter difficulty)

### Version

osu! version: Lazer 2024.817.0 (latest)

### Logs

[compressed-logs.zip](https://github.com/user-attachments/files/16680497/compressed-logs.zip)

**Comment by quat1024** [comment]
This happens in stable as well.

The game increases the size of the fruits before it calculates where to put hyperdashes, so because the fruits are larger, you don't need as many hyperdashes. This tends to turn hyperdashes which were just *barely* over the threshold into edge hits, and yeah, people have made EZ farm maps out of this mechanic, like [this one](https://osu.ppy.sh/beatmapsets/2058169#fruits/4301585).
